nix/tests.nix

```
@@ -102,12 +112,22 @@ in
       virtualisation = {
         cores = 2;
         memorySize = 4096;
+        qemu.options = [
+          # A virtual USB XHCI controller in the host ...
+          "-device qemu-xhci,id=host-xhci,addr=10"
+          # ... with an attached usb stick.
+          "-drive if=none,id=usbstick,format=raw,file=${usbDiskImagePath}"
```

```suggestion
          "-drive if=none,id=usbstick,format=raw,snapshot=on,file=${usbDiskImage}"
```

Does this work to avoid the writable image?

---

This would also unbreak the interactive driver:

```
In [1]: start_all()
start all VMs
machine: starting vm
mke2fs 1.47.2 (1-Jan-2025)
qemu-system-x86_64: -drive if=none,id=usbstick,format=raw,file=/tmp/usb-stick-image: Could not open '/tmp/usb-stick-image': No such file or directory
---------------------------------------------------------------------------
ConnectionResetError                      Traceback (most recent call last)

```

Or at least it would Just Work™ without having to remember to run some magic commands.

---

Yes! That gets rid of the permission denied errors. I'll use that instead and remove the extra tmpfile handling. Thank you!